compiler/codegen/cg-expr.cpp

```diff
@@ -2166,11 +2166,8 @@ GenRet codegenTernary(GenRet cond, GenRet ifTrue, GenRet ifFalse)
 
     func->getBasicBlockList().push_back(blockEnd);
     info->irBuilder->SetInsertPoint(blockEnd);
-#if HAVE_LLVM_VER >= 130
-    ret.val = info->irBuilder->CreateLoad(tmp->getType()->getPointerElementType(), tmp);
-#else
-    ret.val = info->irBuilder->CreateLoad(tmp);
-#endif
+    ret.val = info->irBuilder->CreateLoad(ret.chplType->symbol->getLLVMStructureType(), tmp);
```

Comment:
From what I can tell, `tmp->getType()` would [always be the type of `ifTrue`](https://github.com/chapel-lang/chapel/blob/4897a04842c0b07c833dc80115213967e0503440/compiler/codegen/cg-expr.cpp#L2152), whereas [`ret.chplType` will be the type of `ifFalse` iff `ifTrue`'s type is null](https://github.com/chapel-lang/chapel/blob/4897a04842c0b07c833dc80115213967e0503440/compiler/codegen/cg-expr.cpp#L2122). This is a change in behavior, so unless it is intentional I think instead of `ret.chplType` we want `ifTrue.chplType`.

Reply:
Thanks, I made that change and ran another round of paratest.